### ISSUE TITLE
Hub do not pass `--git-dir` and `--work-tree` flags to original Git

### ISSUE THREAD

**Issue by hauleth** [github/hub#1199]
Versions:

```
git version 2.9.0
hub version 2.2.3
```

This causes issues like tuvistavie/fundle#24 and makes Hub incompatible with original `git` command.

**Comment by mislav** [comment]
Sorry that you're having trouble.

We tried to make sure that these global flags are passed to original `git`. In fact, when I try this command with debug logging, I can see that `--git-dir` and `--work-tree` are passed correctly:

```
$ HUB_VERBOSE=1 hub --git-dir=.git --work-tree=/tmp status

git --git-dir=.git --work-tree=/tmp config alias.status
git --git-dir=.git --work-tree=/tmp status
```

Can you try enabling HUB_VERBOSE and checking if you see anything different?

**Comment by hauleth** [comment]
No, everything looks normal. Cause of the anomaly is still unknown.

**Comment by mislav** [comment]
Well, if `command git` fixed it, then the problem was certainly with hub somehow, and I'd like to track it down. Is it possible that hub was outdated at the time that you ran into this error, or that you have multiple versions of hub installed?

**Comment by rfc1459** [comment]
I can confirm that *some* commands work as intended, but the ones overridden by hub break badly when passing global flags.

Simple reproducer:
```
myuser@myhost:~/somedir$ hub version
git version 2.9.0
hub version 2.2.3
myuser@myhost:~/somedir$ HUB_VERBOSE=1 hub -C somerepo remote
$ git -C somerepo config alias.remote
$ git remote
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
myuser@myhost:~/somedir$ HUB_VERBOSE=1 hub -C somerepo status
$ git -C somerepo config alias.status
$ git -C somerepo status
On branch master
nothing to commit, working directory clean
```

As you can see, `-C somerepo` is correctly propagated to `git status`, but not to `git remote`.

**Comment by mislav** [comment]
@rfc1459 I see. Thanks for the details. I will double-check the `-C` behavior.

**Comment by webbj74** [comment]
I'm running into the same issue on OSX with brew installed git & hub

```
$ git --version
git version 2.9.0
hub version 2.2.3

$ alias git
alias git='hub'
```

```
/tmp $ HUB_VERBOSE=1 git --git-dir=test/.git --work-tree=test checkout 2014-09-08
$ git --git-dir=test/.git --work-tree=test config alias.checkout
$ git --git-dir=test/.git --work-tree=test config hub.host
$ git checkout 2014-09-08
fatal: Not a git repository (or any of the parent directories): .git

/tmp $ /usr/local/bin/git --git-dir=test/.git --work-tree=test checkout 2014-09-08
HEAD is now at 86de88a... adding symlink
```

**Comment by PigeonF** [comment]
I can also confirm the Issue on OSX (installed through homebrew).

```sh
$ hub --version
git version 2.9.2
hub version 2.2.3

$ alias git
git=hub
```

When running commands like `log` or `status`, the flags seem to be passed correctly.  However, when running `push` or `checkout` no flags are passed

Output of `HUB_VERBOSE=1 hub --git-dir=$HOME/.dotfiles.git --work-tree=$HOME push`
```sh
$ git --git-dir=/Users/pigeonf/.dotfiles.git --work-tree=/Users/pigeonf config alias.push
$ git push
fatal: Not a git repository (or any of the parent directories): .git
```

Output of `HUB_VERBOSE=1 hub --git-dir=$HOME/.dotfiles.git --work-tree=$HOME checkout .gitconfig`
```sh
$ git --git-dir=/Users/pigeonf/.dotfiles.git --work-tree=/Users/pigeonf config alias.checkout
$ git --git-dir=/Users/pigeonf/.dotfiles.git --work-tree=/Users/pigeonf config hub.host
$ git checkout .gitconfig
fatal: Not a git repository (or any of the parent directories): .git
```

In addition to the `command git` workaround, it seems that modifying the git directory and work tree using the `GIT_DIR` and `GIT_WORK_TREE` environment variables seems to still work (although this does not fix the underlying issue with hub)

Output of  `HUB_VERBOSE=1 GIT_DIR=$HOME/.dotfiles.git GIT_WORK_TREE=$HOME hub push` 
```sh
$ git config alias.push
$ git push
Everything up-to-date
```

**Comment by mislav** [comment]
This is fixed and going into the 2.2.4 release pending soon. Thanks for all the help!
